### PR TITLE
Flatten subcollections and enhance sweep scope to work with them

### PR DESCRIPTION
**Pull request by Ladas**
Subcollections as concept are being ditched, everything will be flat now.

If we want to archive e.g. container_groups with their 'subcollections' containers, after we save them all we'll send scope: 
```
sweep_scope: ["container_groups", "containers"]
```

If we want to archive it targeted way, so we've saved container_groups X and Y and all their containers we'll send scope:

```
:sweep_scope => {
        :container_groups  => [
          {:ems_ref => "container_group_ems_ref_1"},
          {:ems_ref => "container_group_ems_ref_2"},
        ],
        :containers => [
          {:container_group => persister.container_groups.lazy_find("container_group_ems_ref_1")},
          {:container_group => persister.container_groups.lazy_find("container_group_ems_ref_2")},
        ]
      }
```

The scope provided can contain any attributes on IC.

Depends on: 
- [x] make serialization work
- [x] add `last_seen_on` to every DB table involved in refresh https://github.com/ManageIQ/topological_inventory-core/pull/126
- [ ] https://github.com/ManageIQ/topological_inventory-ingress_api/pull/37
- [ ] https://github.com/ManageIQ/topological_inventory-ingress_api-client-ruby/pull/31

Merge together with:
- [ ] adjust OpenShift collector https://github.com/ManageIQ/topological_inventory-openshift/pull/64 
- [ ] port amazon collector to mark&sweep https://github.com/ManageIQ/topological_inventory-amazon/pull/15

### CODE REVIEW

**Comment by Ladas** [comment]
cc @slemrmartin

File: lib/inventory_refresh/inventory_collection/scanner.rb [hunk 1]
```diff
@@ -136,7 +130,7 @@ def build_parent_inventory_collection!
 
       def supports_building_inventory_collection?
         # Don't try to introspect ICs with custom query or saving code
-        return if arel.present? || custom_save_block.present?
+        return if !arel.nil? || custom_save_block.present?
```
Comment (agrare) [inline lib/inventory_refresh/inventory_collection/scanner.rb]: What is the point of this change?

Comment (Ladas) [reply lib/inventory_refresh/inventory_collection/scanner.rb]: For some reason `arel.present?` started to actually do the query, which can be very expensive. (I was reviewing the sweep queries and found this)

Comment (agrare) [reply lib/inventory_refresh/inventory_collection/scanner.rb]: :scream: okay good change then haha

**Pull request merged**